src/routes/PrimaryLayout/PrimaryLayout.js

```diff
@@ -248,15 +272,18 @@ const createCommunityRoutes = [
 ]
 
 const redirectRoutes = [
-  { from: '/tag/:topicName', to: '/all/topics/:topicName' },
+  { from: '/tag/:topicName', to: '/all/:topicName' },
   { from: '/c/:slug/tag/:topicName', to: '/c/:slug/:topicName' },
   { from: '/c/:slug/join/:accessCode/tag/:topicName', to: '/c/:slug/join/:accessCode/:topicName' },
   { from: '/p/:postId', to: '/all/p/:postId' },
   { from: '/u/:personId', to: '/m/:personId' },
   { from: '/c/:slug/about', to: '/c/:slug' },
-  { from: '/c/:slug/people', to: '/c/:slug/members' },
-  { from: '/c/:slug/invite', to: '/c/:slug/settings/invite' },
-  { from: '/c/:slug/events', to: '/c/:slug' }
+  { from: '/c/:slug/people', to: '/c/:slug/:view(members)' },
+  { from: '/c/:slug/invite', to: '/c/:slug/:view(settings)/invite' },
+  { from: '/c/:slug/events', to: '/c/:slug' },
+  // redirects for context switching into global contexts
```

Comment:
AHHA. It FINALLY got through why this was needed.  :view is obviously not part of the set of things we are clearing on the context switch. Of course. 

Now that I understand what's up better your first solution seems not so bad, but this does feels like a less offensive place to capture (hide perhaps) these exceptions for now--comment included.  Good work~!